snapcraft/internal/parts.py

```diff
@@ -117,6 +124,25 @@ def get_part(self, part_name, full=False):
                 remote_part.pop(key)
         return remote_part
 
+    def matches_for(self, part_match, max_len=0):
```

Comment:
@elopio I am in agreement. An instance of _RemoteParts is created under the implementation of `search` and calls this method so it is not internal to the class https://github.com/ubuntu-core/snapcraft/pull/608/files#diff-ac0a8a66808d29ef0e343cb310093978R179